db/python/layers/billing.py

```diff
@@ -424,3 +509,332 @@ async def get_total_cost(
+                {
+                    'cost_group': 'S' if row.cost_category == 'Cloud Storage' else 'C',
+                    'cost_category': row.cost_category,
+                    'daily_cost': row.daily_cost if is_current_month else None,
+                    'monthly_cost': row.monthly_cost,
+                }
+            )
+
+            total_monthly_category[row.cost_category] += row.monthly_cost
+            if row.daily_cost:
+                total_daily_category[row.cost_category] += row.daily_cost
+
+            # cost groups S/C
+            if row.cost_category == 'Cloud Storage':
+                total_monthly['S']['ALL'] += row.monthly_cost
+                total_monthly['S'][row.field] += row.monthly_cost
+                if row.daily_cost and is_current_month:
+                    total_daily['S']['ALL'] += row.daily_cost
+                    total_daily['S'][row.field] += row.daily_cost
+            else:
+                total_monthly['C']['ALL'] += row.monthly_cost
+                total_monthly['C'][row.field] += row.monthly_cost
+                if row.daily_cost and is_current_month:
+                    total_daily['C']['ALL'] += row.daily_cost
+                    total_daily['C'][row.field] += row.daily_cost
+
+        # add total
+        # construct ALL fields details
+        all_details = []
+        for cat, mth_cost in total_monthly_category.items():
+            all_details.append(
+                {
+                    'cost_group': 'S' if cat == 'Cloud Storage' else 'C',
+                    'cost_category': cat,
+                    'daily_cost': total_daily_category[cat] if is_current_month else None,
+                    'monthly_cost': mth_cost,
+                }
+            )
+
+        # add total row first
+        results.append(
+            BillingCostBudgetRecord.from_json(
+                {
+                    'field': f'{BillingColumn.generate_all_title(field)}',
+                    'total_monthly': (
+                        total_monthly['C']['ALL'] + total_monthly['S']['ALL']
+                    ),
+                    'total_daily': (
+                        total_daily['C']['ALL'] + total_daily['S']['ALL']
+                    ) if is_current_month else None,
+                    'compute_monthly': total_monthly['C']['ALL'],
+                    'compute_daily': (
+                        total_daily['C']['ALL']
+                    ) if is_current_month else None,
+                    'storage_monthly': total_monthly['S']['ALL'],
+                    'storage_daily': (
+                        total_daily['S']['ALL']
+                    ) if is_current_month else None,
+                    'details': all_details,
+                    'last_loaded_day': last_loaded_day,
+                }
+            )
+        )
+
+        # get budget map per gcp project
+        if field == BillingColumn.PROJECT and is_current_month:
+            budget = await self.get_budget()
```

Comment:
Leaving this comment here in case I forget to use it below. This method should be called something like `get_budgets_by_gcp_project` and the variable name should be updated accordingly.